pants-plugins/api_spec/rules.py

```diff
@@ -0,0 +1,259 @@
+    )
+
+    # If we were given an input digest from a previous formatter for the source files, then we
+    # should use that input digest instead of the one we read from the filesystem.
+    source_files_snapshot = (
+        source_files.snapshot if request.snapshot is None else request.snapshot
+    )
+
+    input_digest = await Get(
+        Digest,
+        MergeDigests((dependency_files.snapshot.digest, source_files_snapshot.digest)),
+    )
+
+    result = await Get(
+        ProcessResult,
+        VenvPexProcess(
+            pex,
+            argv=(
+                "--config-file",
+                "conf/st2.dev.conf",
+            ),
+            input_digest=input_digest,
+            description="Regenerating openapi.yaml api spec",
+            level=LogLevel.DEBUG,
+        ),
+    )
+
+    contents = [
+        FileContent(
+            f"{field_set.address.spec_path}/{field_set.source.value}",
+            result.stdout,
+        )
+        for field_set in request.field_sets
+    ]
+
+    output_digest = await Get(Digest, CreateDigest(contents))
+    output_snapshot = await Get(Snapshot, Digest, output_digest)
+    # TODO: Drop result.stdout since we already wrote it to a file?
+    return FmtResult.create(request, result, output_snapshot, strip_chroot_path=True)
+
+
+@rule(
+    desc="Validate openapi.yaml with st2-validate-api-spec",
+    level=LogLevel.DEBUG,
+)
+async def validate_api_spec(
+    request: ValidateAPISpecRequest,
+) -> LintResults:
+    # There will only be one target+field_set, but we iterate
+    # to satisfy how lint expects that there could be more than one.
+    # If there is more than one, they will all get the same contents.
+
+    # Find all the dependencies of our target
+    transitive_targets = await Get(
+        TransitiveTargets,
+        TransitiveTargetsRequest(
+            [field_set.address for field_set in request.field_sets]
+        ),
+    )
+
+    dependency_files_get = Get(
+        SourceFiles,
+        SourceFilesRequest(
+            sources_fields=[
+                tgt.get(SourcesField) for tgt in transitive_targets.dependencies
+            ],
+            for_sources_types=(FileSourceField, ResourceSourceField),
+        ),
+    )
+
+    source_files_get = Get(
+        SourceFiles,
+        SourceFilesRequest(field_set.source for field_set in request.field_sets),
+    )
+
+    # actually validate it with an external script.
+    # Validation cannot be inlined here because it needs to import the st2 code.
+    pex_get = Get(
+        VenvPex,
+        PexFromTargetsRequest(
+            [
+                Address(
+                    CMD_DIR,
+                    target_name="cmd",
+                    relative_file_path=f"{VALIDATE_CMD}.py",
+                ),
+            ],
+            output_filename=f"{VALIDATE_CMD}.pex",
+            internal_only=True,
+            main=EntryPoint.parse(f"{CMD_MODULE}.{VALIDATE_CMD}:main"),
```

Comment:
The same thing applies here. We are using `st2common.cmd.validate_api_spec:main` directly instead of using the `st2-validate-api-spec` script directly.